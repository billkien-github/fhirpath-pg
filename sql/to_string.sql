\set pt $JSON${"resourceType":"Patient", "name": [{"use": "official", "ext": {"bext" : ["fext"]}, "family": ["ivanov", "vodkin"]}, {"use": "common", "family": ["petrov"]}]}$JSON$

SELECT fhirpath_as_string(:pt, '.name', 'HumanName');
SELECT fhirpath_as_string('{"name": [{"use": "SKIP", "period": "SKIP", "family": ["f1", "f2"]}]}', '.name', 'HumanName');
SELECT fhirpath_as_string('{"name": [{"use": "SKIP", "period": "SKIP", "street": ["f1", "f2"]}]}', '.name', 'Address');
SELECT fhirpath_as_string(:pt, '.name.family', 'string');
SELECT fhirpath_as_string(:pt, '.name.where(use=official).family', 'HumanName');
SELECT fhirpath_as_string(:pt, '.name.where(use=common).family', 'HumanName');

SELECT fhirpath_as_string('{"name": [{"use": "SKIP", "period": {"start":  "SKIP", "end": "SKIP"}, "street": ["f1", "f2"]}]}', '.name', 'Address');
