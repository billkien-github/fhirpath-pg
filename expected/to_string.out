\set pt $JSON${"resourceType":"Patient", "name": [{"use": "official", "ext": {"bext" : ["fext"]}, "family": ["ivanov", "vodkin"]}, {"use": "common", "family": ["petrov"]}]}$JSON$
SELECT fhirpath_as_string(:pt, '.name', 'HumanName');
     fhirpath_as_string      
-----------------------------
 $fext$ivanov$vodkin$petrov$
(1 row)

SELECT fhirpath_as_string('{"name": [{"use": "SKIP", "period": "SKIP", "family": ["f1", "f2"]}]}', '.name', 'HumanName');
 fhirpath_as_string 
--------------------
 $f1$f2$
(1 row)

SELECT fhirpath_as_string('{"name": [{"use": "SKIP", "period": "SKIP", "street": ["f1", "f2"]}]}', '.name', 'Address');
 fhirpath_as_string 
--------------------
 $f1$f2$
(1 row)

SELECT fhirpath_as_string(:pt, '.name.family', 'string');
   fhirpath_as_string   
------------------------
 $ivanov$vodkin$petrov$
(1 row)

SELECT fhirpath_as_string(:pt, '.name.where(use=official).family', 'HumanName');
 fhirpath_as_string 
--------------------
 $ivanov$vodkin$
(1 row)

SELECT fhirpath_as_string(:pt, '.name.where(use=common).family', 'HumanName');
 fhirpath_as_string 
--------------------
 $petrov$
(1 row)

SELECT fhirpath_as_string('{"name": [{"use": "SKIP", "period": {"start":  "SKIP", "end": "SKIP"}, "street": ["f1", "f2"]}]}', '.name', 'Address');
 fhirpath_as_string 
--------------------
 $f1$f2$
(1 row)

SELECT fhirpath_as_string('{"a": 222}', '.a', 'Numeric');
 fhirpath_as_string 
--------------------
 $222$
(1 row)

